extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateValidatorRecorder.java

```diff
@@ -121,6 +122,15 @@ public void created(BeanContainer container) {
                 configuration.allowMultipleCascadedValidationOnReturnValues(
                         hibernateValidatorBuildTimeConfig.methodValidation.allowMultipleCascadedValidationOnReturnValues);
 
+                List<InstanceHandle<ConstraintMappingProvider>> constraintMappingProviderList = Arc.container()
+                        .listAll(ConstraintMappingProvider.class);
+                for (InstanceHandle<ConstraintMappingProvider> cmpInstanceHandle : constraintMappingProviderList) {
+                    if (cmpInstanceHandle.isAvailable()) {
+                        final ConstraintMappingProvider constraintMappingProvider = cmpInstanceHandle.get();
+                        constraintMappingProvider.createConstraintMappings(configuration);
+                    }
+                }
```

Comment:
Maybe move it to the very end, just before the call to `configuration.buildValidatorFactory()`. That way, the user can override what was done before if there's a need for it.